paasta_tools/utils.py

```diff
@@ -1567,6 +1571,79 @@ def list_clusters(service=None, soa_dir=DEFAULT_SOA_DIR, instance_type=None):
     return sorted(clusters)
 
 
+def get_instance_configs(instance_type, cluster=None, soa_dir=DEFAULT_SOA_DIR):
+    """Returns instance configs associated with an instance type and cluster
+
+    :param instance_type: The instance type, eg chronos or marathong
+    :param cluster: The cluster to look in.  If not specified will lookup the current cluster
+    :returns: dict in format {service_name: {instance_name: {instance_conf}, ...}, ...}
+    """
+    if not cluster:
+        cluster = load_system_paasta_config().get_cluster()
+    rootdir = os.path.abspath(soa_dir)
+    service_instance_configs = {}
+    for srv_dir in os.listdir(rootdir):
+        service_instance_configs[srv_dir] = service_configuration_lib.read_extra_service_information(
+            srv_dir,
+            "%s-%s" % (instance_type, cluster),
+            soa_dir=soa_dir,
+        )
+    return service_instance_configs
+
+
+def get_all_instance_configs(clusters, soa_dir=DEFAULT_SOA_DIR):
+    """Get all instance configs for all instance types for a list of clusters
+
+    :param clusters: The clusters to get configs for
+    :returns: A dict in the format {cluster_name: {service_name: {instance_name: {instance_conf}, ...}, ...}, ...}
+    """
+    confs = defaultdict(lambda: defaultdict(dict))
+    for t in INSTANCE_TYPES:
+        for c in clusters:
+            services = get_instance_configs(t, c, soa_dir)
+            for service, instance in services.items():
+                confs[c][service].update(instance)
+    return confs
+
+
+def get_all_service_configs(soa_dir=DEFAULT_SOA_DIR):
+    """Get the service config for all services
+
+    :returns: A dict in the format {service_name: {service_conf}, ...}
+    """
+    rootdir = os.path.abspath(soa_dir)
+    service_configs = {}
+    for srv_dir in os.listdir(rootdir):
+        service_configs[srv_dir] = service_configuration_lib.read_service_configuration(
+            srv_dir,
+            soa_dir=soa_dir,
+        )
+    return service_configs
+
+
+def get_instances_by_owners(owners, clusters, soa_dir=DEFAULT_SOA_DIR):
+    """Get all service instances with owner in a list
+
+    :param owners: A list of owners to match against
+    :param clusters: A list of clusters to look for services in
+    :returns: A dict of service_instances, in the format {cluster: {service_name: [instance1, instance2, ...], ...}}
+    """
+    all_instances = {c: get_services_for_cluster(cluster=c, soa_dir=soa_dir) for c in clusters}
+    service_confs = get_all_service_configs(soa_dir)
+    instance_confs = get_all_instance_configs(clusters, soa_dir)
+
+    cluster_services_instances = defaultdict(lambda: defaultdict(set))
+    for cluster, services_instances in all_instances.items():
+        for service, instance in services_instances:
+            service_team = service_confs.get(service, {}).get('monitoring', {}).get('team', None)
```

Comment:
this looks weird to me, are you not able to use the existing monitoring_tools.get_team function? (that does this logic already)